### DISCUSSION TITLE
How to make duplicated action/condition Node with same behavior/algorithm but "InputPort" are unique name for each Node

### DISCUSSION THREAD

**Discussion by aminballoon** [BehaviorTree/BehaviorTree.CPP#612] [Q&A]
I'm using BT4.3.3 and I have tried to make duplicated action or condition Node with same behavior/algorithm and "InputPort" are unique name for each node but the function `providedPorts` is static function it can't set the port with variable . 
Can someone suggest me to solve this issue ?

**Comment by facontidavide** [comment]
I think it COULD be possible with a mix of Polymorphism AND Curiously Recurring Template Pattern, but I can't see a reason why you would wan t to do that.

Can you provide a motivating example? I think you don't need to actually do that.
